src/Core/Algorithms/Legacy/Fields/RegisterWithCorrespondences.cc

```diff
@@ -847,8 +887,50 @@ bool RegisterWithCorrespondencesAlgo::runP(FieldHandle input, FieldHandle Cors1,
       Tmat(2, 1)*traceA*norm1,
       Tmat(2, 2)*traceA*norm1,
       0.0 };
-    
-    //done with solve, make the new field
+  
+  DenseMatrix rotation(Eigen::MatrixXd::Identity(4,4));
+  DenseMatrix trans1(Eigen::MatrixXd::Identity(4,4));
+  DenseMatrix trans2(Eigen::MatrixXd::Identity(4,4));
+  DenseMatrix scale1(Eigen::MatrixXd::Identity(4,4));
+  DenseMatrix scale2(Eigen::MatrixXd::Identity(4,4));
+  double *r_data = rotation.data();
+  
+  for (int p = 0; p < coefs.size(); p++)
+  {
+    r_data[p] = coefs[p];
+  }
```

Comment:
If you must use the data pointer, at least use a copy function to replace the loop:
`std::copy(coefs.begin(), coefs.end(), r_data);`